lib/sshutils/sftp/sftp.go

```diff
@@ -388,15 +389,13 @@ func (c *Config) transferFile(ctx context.Context, dstPath, srcPath string, srcF
 
 	dstFile, err := c.dstFS.Create(ctx, dstPath, srcFileInfo.Mode())
 	if err != nil {
-		if trace.IsAccessDenied(err) && c.opts.Force {
+		if errors.Is(err, fs.ErrPermission) && c.opts.Force {
 			if err := c.dstFS.Remove(ctx, dstPath); err != nil {
 				return trace.Wrap(err)
 			}
 			dstFile, err = c.dstFS.Create(ctx, dstPath, srcFileInfo.Mode())
-			if err != nil {
-				return trace.Wrap(err)
-			}
-		} else {
+		} // Re-check error as it may be nil from the remove-create case above.
+		if err != nil {
```

Comment:
I think I prefer the original block inside the if/right after the Create, but up to you.

Reply:
Oh dear. Sorry @lxea - duelling reviewers are no fun. Mine was just a suggestion, so likewise - up to you.